plugins/check_http.c

```diff
@@ -134,6 +136,33 @@ char buffer[MAX_INPUT_BUFFER];
 char *client_cert = NULL;
 char *client_privkey = NULL;
 
+/*
+ Bitmask of the status codes to yield OK on.
+ 1st bit for 100, 2nd for 101, etc.. None set means defaults.
+*/
+uint64_t status_expect[8] = { 0 };
+
+static inline
+int status_expect_isempty(void)
+{
+  char status_expect_none[sizeof(status_expect)] = { 0 };
+  return !memcmp(&status_expect, &status_expect_none, sizeof(status_expect));
+}
+
+static inline
+void status_expect_set(int code)
+{
+  uint64_t idx = code - 100;
+  status_expect[idx / 64u] |= 1u << (idx % 64u);
+}
+
+static inline
+int status_expect_isset(int code)
+{
+  uint64_t idx = code - 100;
+  return !!(status_expect[idx / 64u] & (1u << (idx % 64u)));
```

Comment:
Please add a comment explaining what this does or rewrite it in an understandable manner.